xc/common/utils/prjxray_arch_import.py

```diff
@@ -116,6 +116,187 @@ def create_synth_io_tiles(complexblocklist_xml, tiles_xml, pb_name, is_input):
     )
 
 
+def create_hetero_synth_io_tiles(complexblocklist_xml, tiles_xml):
+    """ Creates synthetic IO pad tiles used to connect ROI inputs and outputs to the routing network.
+    """
+    pb_xml_in = ET.SubElement(
+        complexblocklist_xml, 'pb_type', {
+            'name': 'SYN-INPAD',
+        }
+    )
+
+    pb_xml_out = ET.SubElement(
+        complexblocklist_xml, 'pb_type', {
+            'name': 'SYN-OUTPAD',
+        }
+    )
+
+    tile_xml = ET.SubElement(tiles_xml, 'tile', {
+        'name': 'SYN-IOPAD',
+    })
+
+    sub_tile_xml_in = ET.SubElement(
+        tile_xml, 'sub_tile', {
+            'name': 'SYN_IN_SUB_TILE',
+            'capacity': '50'
+        }
+    )
+    sub_tile_xml_out = ET.SubElement(
+        tile_xml, 'sub_tile', {
+            'name': 'SYN_OUT_SUB_TILE',
+            'capacity': '50'
+        }
+    )
+
+    equivalent_sites_in = ET.SubElement(sub_tile_xml_in, 'equivalent_sites')
+    site_in = ET.SubElement(
+        equivalent_sites_in, 'site', {'pb_type': 'SYN-INPAD'}
+    )
+    equivalent_sites_out = ET.SubElement(sub_tile_xml_out, 'equivalent_sites')
+    site_out = ET.SubElement(
+        equivalent_sites_out, 'site', {'pb_type': 'SYN-OUTPAD'}
+    )
+
+    ET.SubElement(
+        sub_tile_xml_in, 'fc', {
+            'in_type': 'abs',
+            'in_val': '2',
+            'out_type': 'abs',
+            'out_val': '2',
+        }
+    )
+
+    ET.SubElement(
+        sub_tile_xml_out, 'fc', {
+            'in_type': 'abs',
+            'in_val': '2',
+            'out_type': 'abs',
+            'out_val': '2',
+        }
+    )
+
+    interconnect_xml_in = ET.SubElement(pb_xml_in, 'interconnect')
+    interconnect_xml_out = ET.SubElement(pb_xml_out, 'interconnect')
+
+    blif_model_in = '.input'
+    pad_name_in = 'inpad'
+    port_type_in = 'output'
+
+    blif_model_out = '.output'
+    pad_name_out = 'outpad'
+    port_type_out = 'input'
+
+    ET.SubElement(
+        pb_xml_in, port_type_in, {
+            'name': pad_name_in,
+            'num_pins': '1',
+        }
+    )
+
+    ET.SubElement(
+        pb_xml_out, port_type_out, {
+            'name': pad_name_out,
+            'num_pins': '1',
+        }
+    )
+
+    ET.SubElement(
+        sub_tile_xml_in, port_type_in, {
+            'name': pad_name_in,
+            'num_pins': '1',
+        }
+    )
+
+    ET.SubElement(
+        sub_tile_xml_out, port_type_out, {
+            'name': pad_name_out,
+            'num_pins': '1',
+        }
+    )
+
+    port_pin_in = '{}.{}'.format('SYN-INPAD', pad_name_in)
+    pad_pin_in = '{}.{}'.format(pad_name_in, pad_name_in)
+
+    port_pin_out = '{}.{}'.format('SYN-OUTPAD', pad_name_out)
+    pad_pin_out = '{}.{}'.format(pad_name_out, pad_name_out)
+
+    ET.SubElement(site_in, 'direct', {'from': port_pin_in, 'to': port_pin_in})
```

Comment:
This is probably wrong, because neither port_pin_in nor port_pin_in has indicies.  @acomodi please review too!

Reply:
I believe indices are actually not strictly needed as VPR will handle that if capacity is > 1 for the sub_tile corresponding to this site.

I think also here the `from` should be in the form of `<sub_tile_name>.<pin_name>` to have a valid direct